packages/@aws-cdk/core/lib/names.ts

```diff
@@ -36,5 +38,26 @@ export class Names {
     return components.length > 0 ? makeUniqueId(components) : '';
   }
 
+  /**
+   * Returns a CloudFormation-compatible unique identifier for a construct based
+   * on its path. If the node is a Stack, the stack name is used for that path component instead.
+   * The identifier includes a human readable portion rendered from the path components and a hash suffix.
+   * @param construct The construct
+   * @param maxUniqueIdLength The maximum length of the unique id
+   * @returns a unique id based on the construct path
+   */
+  public static uniqueIdWithStackName(construct: Construct, maxUniqueIdLength: number = 256): string {
```

Comment:
I'd love a less mechanical name for this method, and a more intentional one, but I'm struggling to come up with one.

I'm thinking something like `stackSafeUniqueId`, `environmentallyUniqueId`, ... not coming up with anything great yet. Do you have a good idea for an alternative?

Reply:
I like stackSafeUniqueId so I went with that one.